### DISCUSSION TITLE
Magick++ : Magick::Image::orderedDither() argument

### DISCUSSION THREAD

**Discussion by Pixinn** [ImageMagick/ImageMagick#3149] [Help]
Hi!
I don't understand what argument I have to provide to _Magick::Image::orderedDither_ in order to get the same processing as with the _-ordered-dither_ option in command line.
Whatever I try, I get this error:
> "Magick: InvalidArgument ordered-dither : '' @ error/threshold.c/OrderedPosterizeImageChannel/1477"

I tried "o3x3,2", "o3x3", "3x3" with no success.

**Comment by dlemstra** [comment]
It should be the same argument as on the command line. Can you provide the version that you are using, a code sample and the image you are using to test this?

**Reply by Pixinn** [reply]
Sure.
I'm using version 6.9.2-5.
Here is a code snippet:
> imageRGB.quantizeColors(nbColors);
> imageRGB.quantizeDither(dither);
> imageRGB.quantizeDitherMethod(Magick::FloydSteinbergDitherMethod);
> imageRGB.orderedDither("o3x3,2");
> imageRGB.quantize();`

imageRGB being a _Magick::Image_.

Only the 4th line is problematic, as stated in the question.

**Comment by dlemstra** [comment]
What happens when you upgrade to a more recent version?

**Reply by Pixinn** [reply]
Same thing with version 6.9.11-54. Of course the code emitting the error is not at the same line:
> error: Magick: InvalidArgument ordered-dither : 'o3x3,2' @ error/threshold.c/OrderedPosterizeImageChannel/1888

I cannot easily move to IM7.
